include/miral/miral/wayland_extensions.h

```diff
@@ -33,6 +33,14 @@ namespace mir { class Server; }
 
 namespace miral
 {
+
+/// Enables shell components such as panels, notifications and lock screens. Malicious clients could potentially
+/// use this protocol to steal input focus or otherwise bother the user
+static char constexpr zwlr_layer_shell_v1[] = "zwlr_layer_shell_v1";
```

Comment:
Injecting a character array into every translation unit that uses this header is an interesting way to avoid ABI breaks if we ever drop a protocol extension.

I'm not sure I like it (but open to persuasion). Only dropping extension when we break ABI doesn't seem an undue burden.

Reply:
Seems reasonable to only drop extensions on an ABI break, and then we can throw if an unsupported extension is requested. I'll put the strings in the cpp file and this in symbols.map.